src/workerd/io/worker-fs.c++

```diff
@@ -695,13 +703,22 @@ class FileImpl final: public File {
     }
   }
 
-  kj::Rc<File> clone(jsg::Lock&) override {
+  kj::OneOf<FsError, kj::Rc<File>> clone(jsg::Lock& js) override {
+    auto maxSize = Worker::Isolate::from(js).getLimitEnforcer().getBlobSizeLimit();
     KJ_SWITCH_ONEOF(ownedOrView) {
       KJ_CASE_ONEOF(owned, kj::Array<kj::byte>) {
-        return kj::rc<FileImpl>(kj::heapArray<kj::byte>(owned));
+        if (owned.size() > maxSize) [[unlikely]] {
```

Comment:
I don't get this case, if is is bigger than maxSize then the isolate is condemned and presumably it's not bigger by a lot as that would cause termination, feels unnecessary to add this check here.

Reply:
It generally should never happen (which is why it's marked explicitly `[[unlikely]]`... but good as an additional check in case regressions are introduced at some time.

Reply:
the node.js and WebFS APIs have different expectations for errors. We cannot really just use JSG_REQUIRE's in these lower bits as that makes ensuring the appropriate errors are thrown via each of the APIs rather more cumbersome.